### PR TITLE
 [Juniper] Split Juniper JunOS datastream into separate package  

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Split Juniper JunOS datastream into separate package

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## How to test this PR locally
```
cd packages/juniper_junos
elastic-package build && elastic-package stack down && elastic-package stack up --version 7.16.0-SNAPSHOT -d -v && eval "$(elastic-package stack shellinit)" && elastic-package test -v
```

## Related issues

- Relates #1934

### CODE REVIEW

**Comment by P1llus** [comment]
@legoguy1000 Could you bump this to 8.0?

File: packages/juniper/manifest.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 format_version: 1.0.0
 name: juniper
 title: Juniper
-version: 1.0.3
-description: Collect logs from Juniper devices with Elastic Agent.
+version: 1.0.4
+description: Deprecated. Use a specific Juniper package instead.
```
Comment (legoguy1000) [inline packages/juniper/manifest.yml:5]: I have to remove this. Got committed by accident.

**Comment by P1llus** [comment]
Would you be able to generate a system test output file as well? Since these are only parsed from local scripting the pipeline tests won't show much.

You can use this while the stack is running, then attach the sample file to the PR as well:
`elastic-package test system --generate -v`

**Review by P1llus — APPROVED** [review]

**Pull request merged**